File: leetcode/swap-nodes-in-pairs.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var swapPairs = function(head) {
    return swap(head);
}
function swap(node) {
    if(node === null || node.next === null) return node;
    const temp = node.next;
    node.next = temp.next;
    temp.next = node;
    node.next = swap(node.next);
    return temp;
}

File: leetcode/swap-nodes-in-pairs.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def swapPairs(self, head: ListNode) -> ListNode:
        if head is None or head.next is None:
            return head
        
        temp = head.next
        head.next = temp.next
        temp.next = head
        head.next = self.swapPairs(head.next)
        
        return temp
